tests/test_observation_models.py

```diff
@@ -1001,3 +1012,324 @@ def test_repr_out(self, link_func, link_func_name):
+        )
+        if (pseudo_r2 > 1) or (pseudo_r2 < 0):
+            raise ValueError(f"pseudo-r2 of {pseudo_r2} outside the [0,1] range!")
+
+    @pytest.mark.parametrize("score_type", ["pseudo-r2-Cohen", "pseudo-r2-McFadden"])
+    def test_pseudo_r2_mean(self, score_type, negativeBinomialGLM_model_instantiation):
+        _, y, model, _, _ = negativeBinomialGLM_model_instantiation
+        pseudo_r2 = model.observation_model.pseudo_r2(
+            y, y.mean(), score_type=score_type
+        )
+        if not np.allclose(pseudo_r2, 0, atol=1e-6, rtol=0.0):
+            raise ValueError(
+                f"pseudo-r2 of {pseudo_r2} for the null model. Should be equal to 0!"
+            )
+
+    def test_emission_probability(self, negativeBinomialGLM_model_instantiation):
+        _, _, model, _, _ = negativeBinomialGLM_model_instantiation
+        key_array = jax.random.key(123)
+        gkey, pkey = jax.random.split(key_array)
+        p = np.random.rand(10)
+        counts = model.observation_model.sample_generator(
+            key_array, p, scale=model.observation_model.scale
+        )
+        r = 1.0 / model.observation_model.scale
+        gamma_sample = jax.random.gamma(gkey, r, shape=p.shape) * (p / r)
+        expected_counts = jax.random.poisson(pkey, gamma_sample)
+        if not jnp.allclose(counts, expected_counts):
+            raise ValueError(
+                "The emission probability doesn't match expected NB sampling."
+            )
+
+    @pytest.mark.parametrize(
+        "score_type, expectation",
+        [
+            ("pseudo-r2-McFadden", does_not_raise()),
+            (
+                "not-implemented",
+                pytest.raises(
+                    NotImplementedError, match="Score not-implemented not implemented"
+                ),
+            ),
+        ],
+    )
+    def test_not_implemented_score(
+        self, score_type, expectation, negativeBinomialGLM_model_instantiation
+    ):
+        _, y, model, _, firing_rate = negativeBinomialGLM_model_instantiation
+        with expectation:
+            model.observation_model.pseudo_r2(y, firing_rate, score_type)
+
+    @pytest.mark.parametrize(
+        "scale, expectation",
+        [
+            (0.1, does_not_raise()),
+            (
+                "invalid",
+                pytest.raises(
+                    ValueError, match="The `scale` parameter must be of numeric type"
+                ),
+            ),
+        ],
+    )
+    def test_scale_setter(
+        self, scale, expectation, negativeBinomialGLM_model_instantiation
+    ):
+        _, _, model, _, _ = negativeBinomialGLM_model_instantiation
+        with expectation:
+            model.observation_model.scale = scale
+
+    def test_scale_getter(self, negativeBinomialGLM_model_instantiation):
+        _, _, model, _, _ = negativeBinomialGLM_model_instantiation
+        assert model.observation_model.scale == 1  # or your default value
```

Comment:
```suggestion
        assert model.observation_model.scale == 1  
```
doesn't add anything